### ISSUE TITLE
[MultiSelectTree] Heterogenous Data example is broken

### ISSUE THREAD

**Issue by mbechev** [telerik/kendo-angular#3323]
**Describe the bug**
The example from [this article](https://www.telerik.com/kendo-angular-ui-develop/components/dropdowns/multiselecttree/data-binding/#toc-heterogenous-data) is broken. When selecting a child item from the first parent, the second parent gets selected as well, due to the different fields passed to the **textField** and **valueField** properties:
```html
[textField]="['categoryName', 'subCategoryName']"
[valueField]="['categoryId', 'subCategoryId']"
```

Refer to the following screencast:
https://screenrec.com/share/B8IV3uKw9l

The example works as expected if **valueField** array includes the same fields as **textField** input:
```html
[textField]="['categoryName', 'subCategoryName']"
[valueField]="['categoryName', 'subCategoryName']"
```

**To Reproduce**
https://stackblitz.com/edit/angular-zh76zs

**Comment by ipeshev** [comment]
Fix available in [latest dev version](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds).

**Comment by tapopov** [comment]
Shipped in v6.0.2
